rclpy/src/rclpy/clock.cpp

```diff
@@ -34,108 +34,68 @@ using pybind11::literals::operator""_a;
 
 namespace rclpy
 {
-/// Destructor for a clock
-static void
-_rclpy_destroy_clock(void * p)
+Clock::Clock(rcl_clock_type_t clock_type)
 {
-  auto clock = static_cast<rcl_clock_t *>(p);
-  if (!clock) {
-    // Warning should use line number of the current stack frame
-    int stack_level = 1;
-    PyErr_WarnFormat(
-      PyExc_RuntimeWarning, stack_level, "Failed to get clock pointer in destructor");
-    rcl_reset_error();
-    return;
-  }
-
-  rcl_ret_t ret_clock = rcl_clock_fini(clock);
-  PyMem_Free(clock);
-  if (ret_clock != RCL_RET_OK) {
-    // Warning should use line number of the current stack frame
-    int stack_level = 1;
-    PyErr_WarnFormat(
-      PyExc_RuntimeWarning, stack_level, "Failed to fini clock: %s",
-      rcl_get_error_string().str);
-    rcl_reset_error();
-  }
-}
-
-py::capsule
-create_clock(rcl_clock_type_t clock_type)
-{
-  auto deleter = [](rcl_clock_t * ptr) {_rclpy_destroy_clock(ptr);};
-  auto clock = std::unique_ptr<rcl_clock_t, decltype(deleter)>(
-    static_cast<rcl_clock_t *>(PyMem_Malloc(sizeof(rcl_clock_t))),
-    deleter);
-  if (!clock) {
-    throw std::bad_alloc();
-  }
+  // Create a client
+  rcl_clock_ = std::shared_ptr<rcl_clock_t>(
+    new rcl_clock_t,
+    [this](rcl_clock_t * clock)
```

Comment:
No need to capture `this` in the lambda because it only uses the passed `clock` pointer